.travis.yml

```diff
@@ -77,37 +77,40 @@ jobs:
       after_success:
       - codecov
 
-    - stage: e2e
+    - stage: "Test"
       name: "containerd - calico e2e Tests"
       language: python
       cache: pip
       python: "3.8"
       env:
       - KUBECONFIG="$HOME/.kube/config"
       - CONTAINERD_VERSION=1.3.4
-      # TODO upgrade to v1.18.6 or newer
-      - KUBERNETES_VERSION=v1.15.3
-      # TODO update to v1.12.1 or newer
-      - MINIKUBE_VERSION=v1.3.1
+      # stable version in minikube v1.12.1
+      - KUBERNETES_VERSION=v1.18.3
+      - MINIKUBE_VERSION=v1.12.1
+      - DOCKER_REGISTRY=localhost:5000
       - CHANGE_MINIKUBE_NONE_USER=true
       - MINIKUBE_WANTUPDATENOTIFICATION=false
       - MINIKUBE_WANTREPORTERRORPROMPT=false
       - MINIKUBE_HOME=$HOME
+      - CONTAINER_RUNTIME_ENDPOINT=unix:///run/containerd/containerd.sock
+      - IMAGE_SERVICE_ENDPOINT=unix:///run/containerd/containerd.sock
+      - CRICTL_VERSION="v1.17.0"
 
       before_install:
       # See: https://github.com/containerd/cri/blob/master/docs/installation.md
-      - sudo apt-get update && sudo apt-get install -y libseccomp2 conntrack
-      - curl --fail -sLO "https://storage.googleapis.com/cri-containerd-release/cri-containerd-${CONTAINERD_VERSION}.linux-amd64.tar.gz"
-      - sudo tar --no-overwrite-dir -C / -xzf cri-containerd-${CONTAINERD_VERSION}.linux-amd64.tar.gz
+      - sudo apt-get update && sudo apt-get install -y libseccomp2 conntrack containerd
+      - wget https://github.com/kubernetes-sigs/cri-tools/releases/download/${CRICTL_VERSION}/crictl-${CRICTL_VERSION}-linux-amd64.tar.gz &&
+        sudo tar zxvf crictl-${CRICTL_VERSION}-linux-amd64.tar.gz -C /usr/local/bin &&
+        rm -f crictl-${CRICTL_VERSION}-linux-amd64.tar.gz
 
       before_script:
       # install dependent binaries
       - curl --fail -Lo kubectl "https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubectl" && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
       - curl --fail -Lo minikube "https://storage.googleapis.com/minikube/releases/${MINIKUBE_VERSION}/minikube-linux-amd64" && chmod +x minikube && sudo mv minikube /usr/local/bin/
       - mkdir -p $HOME/.kube $HOME/.minikube
       - touch $KUBECONFIG
-      # todo switch to kind !
-      - minikube config set vm-driver none
+      - minikube config set driver none
```

Comment:
I think instead of the `none` driver we should use the `docker` driver, this should make most things easier and a quick prototype was pretty promising.

Reply:
Migrated to `docker` driver